### PR TITLE
sig-storage: fix a flake where we never ensure PVC is gone

### PR DESCRIPTION
**Pull request by akalenyu**
### What this PR does
This came up with 1.31 lanes more often but is normally an issue
all across the board, since we never ensure the PVC is gone before testing that scenario.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/storage/datavolume.go [hunk 1]
```diff
@@ -955,6 +955,7 @@ var _ = SIGDescribe("DataVolume Integration", func() {
 				}
 				err = virtClient.CoreV1().PersistentVolumeClaims(testsuite.NamespaceTestAlternative).Delete(context.Background(), dataVolume.Name, metav1.DeleteOptions{})
```
Comment (ShellyKa13) [inline tests/storage/datavolume.go:956]: how come we need to delete the pvc that is created by the dv? isnt it enough to wait for the pvc to be deleted?

Comment (akalenyu) [reply tests/storage/datavolume.go]: yep, good point, added a change

**Review by ShellyKa13 — APPROVED** [review]
Cool!
/lgtm

**Pull request merged**